File: Tests/Extensions/LinesTests.fs
namespace Tests.Extensions

open System
open System.IO
open Stringier.Extensions
open Microsoft.VisualStudio.TestTools.UnitTesting

[<TestClass>]
type LinesTests() =
    [<TestMethod>]
    member _.``lines`` () =
        let source = "Some example text." + Path.DirectorySeparatorChar.ToString() + "With some line breaks." + Path.DirectorySeparatorChar.ToString() + "Which should be split up."
        CollectionAssert.AreEqual([|"Some example text.";"With some line breaks.";"Which should be split up."|], lines source)

File: Tests/LinesTests.cs
using System;
using Stringier;
using Defender;
using Xunit;

namespace Tests {
	public class LinesTests : Trial {
		[Theory]
		[InlineData("Some example text.\nWith some line breaks.\nWhich will be split up", new[] { "Some example text.", "With some line breaks.", "Which will be split up"})]
		public void Lines(String source, String[] expected) => Claim.That(source.Lines()).SequenceEquals(expected);
	}
}
